src/context.ts

```diff
@@ -43,12 +43,12 @@ export class Context {
         const locations = _.uniqBy(instances, 'location');
 
         // #331 - When multiple locations are involved, all of them must pass the validation.
-        // If none of the locations contain the field, we at least include one for error reporting.
+        // If none of the locations contain the field, we should check all locations since we are not sure which location the consumer uses.
         // #458, #531 - Wildcards are an exception though: they may yield 0..* instances with different
         // paths, so we may want to skip this filtering.
         if (instances.length > 1 && locations.length > 1 && !group.includes('*')) {
           const withValue = instances.filter(instance => instance.value !== undefined);
-          return withValue.length ? withValue : [instances[0]];
+          return withValue.length ? withValue : instances;
```

Comment:
This imply that even locations where `!Object.prototype.hasOwnProperty(location, param)` get the default value added🤔

Reply:
Hi buddy, thanks for the review. :D 
Just for clarification, the line I changed (Line 51) used to make the `default` only works for `req.body` (which is the first item of `instances`), and I thought maybe it would probably makes more sense to make it works for all `location`s when consumer hasn't set any values for `in`.

So in the commit, I changed to `instances` and this makes the `default` feature in `checkSchema` to be more consistent with the other scenario that this lib currently supports already. [here](https://express-validator.github.io/docs/sanitization-chain-api.html#defaultdefault_value)

If we set up like 
```
app.use('/user', query('greeting').default('hiya'), userController);
```
and send request `GET /user`, the `req.query` will be transformed into `{ greeting: 'hiya' }`

So in short, like you commented, `!Object.prototype.hasOwnProperty(location, param)` does get the default value added... and this behavior is consistent with the other API [here](https://express-validator.github.io/docs/sanitization-chain-api.html#defaultdefault_value)

Reply:
Well #1070 is actually not related to this PR as written in the issue.
However the first problem that comes to my mind is that line 51 is not only used by `.default` and is shared across all the other validators and sanitizers. The fix may be quite though sadly :(

Reply:
Got you, thanks for maintain the lib ! Glad to add more tests if can help save some verification effort (maybe another PR).